src/WidgetController.types.ts

```diff
@@ -103,3 +106,27 @@ export enum SwapWidgetBorderRadiusSize {
   Medium = "medium",
   Large = "large"
 }
+
+export type GenerateWidgetIframeUrlBaseParams = {
+  network?: NetworkToggleValue;
+  /** theme variables to customize the UI of the widget */
+  themeVariables?: SwapWidgetThemeVariables;
+  /** when provided, messages will be posted only to this origin.
+   * @example ```{parentUrlOrigin: "http://localhost:3001}```
+   */
+  parentUrlOrigin?: string;
+  /**
+   * the asset ids to be used for the swap.
+   * order: [assetInId, assetOutId]
+   */
+  assetIds?: [number, number];
+};
+
+export type GenerateWidgetIframeUrlParams =
+  | (GenerateWidgetIframeUrlBaseParams & {
+      useParentSigner: true;
+      accountAddress: string;
```

Comment:
```suggestion
      /** address of the signer. will be used in transactions, and to show account data in the widget */
      accountAddress: string;
```
can be nice to add a short description